src/full/Agda/Syntax/Translation/AbstractToConcrete.hs

```diff
@@ -1538,10 +1548,23 @@ instance ToConcrete A.Pattern where
       let funCtx = applyUnless (null args2) (withPrecedence FunctionCtx)
       tryToRecoverPatternSynP (f args) $ funCtx (tryToRecoverOpAppP $ f args1) >>= \case
         Just c  -> applyTo args2 c
-        Nothing -> applyTo args . C.IdentP =<< toConcrete x
+        Nothing -> applyTo args . C.IdentP True =<< toConcrete x
     -- Note: applyTo [] c = return c
     applyTo args c = bracketP_ (appBracketsArgs args) $ do
-      foldl C.AppP c <$> toConcreteCtx argumentCtx_ args
+      foldl C.AppP c <$>
+        (mapM avoidPun =<< toConcreteCtx argumentCtx_ args)
+
+    -- If --hidden-argument-puns is active, then {x} is replaced by
+    -- {(x)} and ⦃ x ⦄ by ⦃ (x) ⦄.
+    avoidPun :: NamedArg C.Pattern -> AbsToCon (NamedArg C.Pattern)
+    avoidPun arg =
+      ifM (optHiddenArgumentPuns <$> pragmaOptions)
```

Comment:
To avoid `ifM`, you can join the `<&>` fashion trend:
```haskell
pragmaOptions <&> optHiddenArgumentsPuns <&> \case
  True -> case arg of
     ...
  False -> arg
```